File: domusviewer/app/src/main/java/it/achdjian/paolo/domusviewer/zigbee/ZDevice.java
package it.achdjian.paolo.domusviewer.zigbee;

import java.util.Collection;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

import it.achdjian.paolo.domusviewer.DomusEngineRest.JSonDevice;

/**
 * Created by Paolo Achdjian on 15/04/16.
 */
public class ZDevice {
    public int short_address;
    public String extended_address;
    public int capability;
    public Set<Integer> requestingEndpoint = new HashSet<>();
    public Map<Integer,ZEndpoint> endpoints = new HashMap<>();

    public ZDevice(JSonDevice json) {
        short_address = json.short_address;
        extended_address = json.extended_address;
        requestingEndpoint.addAll(json.endpoints.values());
        capability = json.capability;
    }

    public ZDevice(){
    }

    boolean hasEndpoint(Integer endpoint){
        for (Integer containedEndpoint  : endpoints.keySet()) {
            if (containedEndpoint.equals(endpoint)){
                return true;
            }
        }
        return false;
    }

    public void merge(JSonDevice zDevice) {
        requestingEndpoint.addAll(zDevice.endpoints.values());
    }

    public Collection<ZEndpoint> getEndpoins(){
        return endpoints.values();
    }

    public ZEndpoint getEndpoin(int endpoint) {
        return endpoints.get(endpoint);
    }
}

File: temperature_monitor/app/src/main/java/it/achdjian/paolo/temperaturemonitor/zigbee/ZDevice.kt
package it.achdjian.paolo.temperaturemonitor.zigbee

import it.achdjian.paolo.temperaturemonitor.domus_engine_rest.JsonDevice

/**
 * Created by Paolo Achdjian on 7/9/17.
 */
class ZDevice(val shortAddress: Int, val extendedAddr: String, val capabilities: Int) {
    val endpoints: MutableMap<Int, ZEndpoint?> = HashMap()

    constructor(json: JsonDevice):this(json.short_address, json.extended_address, json.capability){
        for (value in json.endpoints.values) {
            endpoints.put(Integer.parseInt(value, 16),null)
        }
    }

    fun hasEndpoint(endpoint: Int): Boolean {
        for (containedEndpoint in endpoints.keys) {
            if (containedEndpoint == endpoint && endpoints[endpoint] != null) {
                return true
            }
        }
        return false
    }
}